beacon_node/lighthouse_network/src/service/mod.rs

```diff
@@ -1103,22 +1103,21 @@ impl<E: EthSpec> Network<E> {
             .sync_committee_bitfield::<E>()
             .expect("Local discovery must have sync committee bitfield");
 
-        {
```

Comment:
no need to open another scope, we can just `drop` the write lock ahead and cache `local_metadata` to save another `read` call